src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,216 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null && tail == null && size == 0) {
+            addToEmptyList(value);
+        } else if (head != null && tail == null) {
+            addInside(value);
+        } else {
+            addIfLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        isPositionIndexBigger(index);
+        if (size == 0) {
+            add(value);
+            return;
+        }
+        if (index == 0 && size != 0) {
+            addAtFirstPosition(value);
+        } else if (index == size) {
+            addOnLastPositionWithIndex(value);
+        } else {
+            addInsideByIndex(index, value);
+        }
```

Comment:
1. code can be more simple
2. naming improvements
```suggestion
        if (index == 0) {
            addAtFirstPosition(value);
            return;
        } 
        else if (index == size) {
            addOnLastPosition(value);
            return;
        } 
        addByIndex(index, value);
```